core/core.go

```diff
@@ -969,6 +969,12 @@ func (t *TextileNode) registerGatewayHandler() {
 		}
 	}()
 	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		cookie, err := r.Cookie("SessionId")
+		if err != nil || cookie.Value != t.GatewayPassword {
+			w.WriteHeader(401)
+			return
+		}
+		log.Infof("valid cookie: %s\n", cookie.Value)
```

Comment:
nice! only thing here... a program could be listening to our logs and pick out this cookie value. next PR just remove it from logging